internal/scanner/walk.go

```diff
@@ -52,6 +53,8 @@ type Walker struct {
 	TempLifetime time.Duration
 	// If CurrentFiler is not nil, it is queried for the current file before rescanning.
 	CurrentFiler CurrentFiler
+	// If MtimeRepo is not nil, it is used to provide mtimes on systems that don't support setting arbirtary mtimes.
+	MtimeRepo *db.VirtualMtimeRepo
```

Comment:
As far as I understand it will never be nil? I'd be in favor of it sometimes being nil and skipping the overhead of this on systems that support chtimes. But that's just personal preference.